portfolio/src/main/webapp/assets/script.js

```diff
@@ -43,3 +18,126 @@ function collapseMenu() {
         });
     }
 }
+
+/**
+ * Opens and closes the navigation bar.
+ */
+
+function openNav() {
+    document.getElementById("nav").style.width = "200px";
+    document.getElementById("nav-main").style.marginLeft = "200px";
+}
+
+function closeNav() {
+    document.getElementById("nav").style.width = "0px";
+    document.getElementById("nav-main").style.marginLeft = "0px";
+}
+
+/** 
+ * Buttons for photo slider transitions, indexers.
+ */
+
+var index = 1;
+
+function move(n) {
+    show(index += n);
+}
+
+function current(n) {
+    show(index = n);
+}
+
+function show(n) {
+    var i;
+    var slides = document.getElementsByClassName("slides");
+    var indices = document.getElementsByClassName("index");
+
+    if (n > slides.length) {
+        index = 1;
+    }
+
+    if (n < 1) {
+        index = slides.length;
+    }
+
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";
+    }
+
+    for (i = 0; i < slides.length; i++) {
+        indices[i].className = indices[i].className.replace(" active", "");
+    }
+
+    if (slides[index - 1] != null) {
+        slides[index - 1].style.display = "block";
+    }
+
+    if (indices[index - 1] != null) {
+        indices[index - 1].className += " active";
+    }
```

Comment:
It'd be best to treat the `index` variable as an actual ordinal number, instead of subtracting 1 from it over here